applications/MedApplication/custom_io/med_model_part_io.cpp

```diff
@@ -440,10 +567,12 @@ void MedModelPartIO::ReadModelPart(ModelPart& rThisModelPart)
         MED_CONNECTIVITY, MED_NODAL,
         &coordinatechangement, &geotransformation); // TODO error if smaller zero, holds probably for the other functions too that return med_int
 
-    int num_geometries_total = 0;
+    IndexType num_geometries_total = 0;
+
+    std::unordered_map<std::string, std::vector<IndexType>> smp_geoms;
 
     // looping geometry types
-    for (int it_geo=1; it_geo<=num_geometry_types; ++it_geo) {
+    for (int it_geo=1; it_geo<=num_geometry_types; ++it_geo) { //TODO +1?
```

Comment:
soo
is it 1-based or not?

Reply:
good catch, I think the current version works correct, but I will check it again

Reply:
works correct as is, removed the comment